### DISCUSSION TITLE
[v4] Remove unused css variables

### DISCUSSION THREAD

**Discussion by madmoizo** [tailwindlabs/tailwindcss#15433] [Ideas]
I'm sure it will be built-in at some point but for now if someone wants to optimize the css output by removing unused css var, here is a little vite plugin I made:
https://github.com/madmoizo/vite-plugin-tailwindcss-cleaner

**Comment by iquito** [comment]
With v4 the amount of unused css variables have gone up by a lot - for my project just those unused variables are about 31% of the total CSS (12kb of 38kb, minified build), I have been removing those 12kb with https://github.com/tomasklaen/postcss-prune-var so far but it would be nice if TailwindCSS could provide some native optimizations to reduce this, as even with the remaining code I am sure a lot of it is probably unnecessary/could be removed for my projects: In v3 the total CSS size for my project was 22kb (with no removing of unused variables or any other special optimizations outside of Tailwind), in v4 it went up to 38kb and even with unused variables removed it is at 26kb.

**Reply by madmoizo** [reply]
There isn't more than [~13kb](https://github.com/madmoizo/vite-plugin-tailwindcss-cleaner?tab=readme-ov-file#is-it-worth-it) of dead code, so it's probably the best you can save. 
If the output is larger in v4, I bet it's because of the use of `@property` (it takes ~3kb in my project)

**Comment by tyler36** [comment]
Thank you for sharing @madmoizo 
Works well with Laravel + Vite setup

Before: 17.29
After: 4.04

**Comment by madmoizo** [comment]
This feature is supported since Tailwind CSS v4.0.5
